src/projects/trio/trio.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+
+#include <array>
+#include <vector>
+#include <unordered_map>
+#include <sequences/seqio.hpp>
+#include "common/logging.hpp"
+#include <common/string_utils.hpp>
+#include "lja/multi_graph.hpp"
+#include "lja/subdataset_processing.hpp"
+
+
+
+struct HaplotypeRemover {
```

Comment:
Why is it not a class. Do all its fields need to be public?

Reply:
> 1. Any haplo logic should be moved out of multigraph. Use delegation, inheritance or whatever but this must be done.
There is no used haplo logic, I've just didn't cleaned every mention of it. Fixing the remains.

> 2. I do not see any reason to use labels is unique identificators of edges and vertices. This overcomplicates code. You can just keep them as extra info inside vertices and edges for meaningful output.
This is reasonable, I even had a todo to move to IDs. Do you think it is necessary before merging?

> 3. Multigraph seems to be broken with respect to some difficult cases like self-rc vertices, edges and loops.
OK

> 4. It is quite important to have a graph as a final output. Let's discuss what can be done.
Answered below. We have hpc graph after trio, and do not output graph in LJAPolish. This is not a trio issue.
> 5. It is strange that you perform your complete analysis twice. It seems like only edge deleting should be performed twice.
There is nearly nothing except edge deletion and graph simplification. Bulge classification can be done once, but it takes not even seconds.

Reply:
This is a data-storage only object, I used to use struct for those. Yes, I believe that fields should better be public here

Reply:
Ok, let it be like this for now. i will look at this some other time. in general it is strange that class that performs some algorithms also works as a free access data storage. These are two different functions that are normally distributed between several classes.